New php stan issues need to be resolved

Resolve the following new issues detected by `phpstan`:

```
Note: Using configuration file /home/darling/Git/PHPReflectionUtilities/phpstan.neon.
 10/10 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ------ ----------------------------------------------------------------------------------------
  Line   src/classes/utilities/Reflection.php
 ------ ----------------------------------------------------------------------------------------
  637    Call to an undefined method ReflectionIntersectionType|ReflectionNamedType::getName().
 ------ ----------------------------------------------------------------------------------------

 ------ ----------------------------------------------------------------------------------------
  Line   tests/interfaces/utilities/ReflectionTestTrait.php (in context of class
         Darling\PHPReflectionUtilities\Tests\classes\utilities\ObjectReflectionTest)
 ------ ----------------------------------------------------------------------------------------
  422    Call to an undefined method ReflectionIntersectionType|ReflectionNamedType::getName().
 ------ ----------------------------------------------------------------------------------------

 ------ ------------------------------------------------------------------------------------------------------------------------------------------------
  Line   tests/interfaces/utilities/ReflectionTestTrait.php (in context of class Darling\PHPReflectionUtilities\Tests\classes\utilities\ReflectionTest)
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------
  422    Call to an undefined method ReflectionIntersectionType|ReflectionNamedType::getName().
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------


 [ERROR] Found 3 errors

```

--- a/.gitignore
+++ b/.gitignore
@@ -1,3 +1,4 @@
 /vendor/
 .phpunit.result.cache
 .phpunit.cache/
+.phpactor.json

--- a/src/classes/utilities/Reflection.php
+++ b/src/classes/utilities/Reflection.php
@@ -625,20 +625,21 @@ class Reflection implements ReflectionInterface
      *
      */
     private function addUnionTypesToArray(
-        ReflectionProperty|ReflectionParameter
-        $parameterOrProperty,
+        ReflectionProperty|ReflectionParameter $parameterOrProperty,
         array &$types,
         ReflectionUnionType $reflectionUnionType
     ): void
     {
         $reflectionUnionTypes = $reflectionUnionType->getTypes();
         foreach($reflectionUnionTypes as $unionType) {
-            $types[$parameterOrProperty->getName()][]
-                = $unionType->getName();
-            $types[$parameterOrProperty->getName()] =
-                array_unique(
-                    $types[$parameterOrProperty->getName()]
-                );
+            if($unionType instanceof ReflectionNamedType) {
+                $types[$parameterOrProperty->getName()][]
+                    = $unionType->getName();
+                $types[$parameterOrProperty->getName()] =
+                    array_unique(
+                        $types[$parameterOrProperty->getName()]
+                    );
+            }
         }
         if(
             !in_array(

--- a/tests/interfaces/utilities/ReflectionTestTrait.php
+++ b/tests/interfaces/utilities/ReflectionTestTrait.php
@@ -329,7 +329,7 @@ trait ReflectionTestTrait
      *                                           by the specified
      *                                           $reflectionUnionType.
      *
-     * @param array<string, array<int, string>> &$types
+     * @param array<string, array<int, string>> $types
      *                                             The array of $types
      *                                             to add the array of
      *                                             strings indicating
@@ -410,20 +410,21 @@ trait ReflectionTestTrait
      *
      */
     private function addUnionTypesToArray(
-        ReflectionProperty
-        |ReflectionParameter $parameterOrProperty,
+        ReflectionProperty|ReflectionParameter $parameterOrProperty,
         array &$types,
         ReflectionUnionType $reflectionUnionType
     ): void
     {
         $reflectionUnionTypes = $reflectionUnionType->getTypes();
         foreach($reflectionUnionTypes as $unionType) {
-            $types[$parameterOrProperty->getName()][]
-                = $unionType->getName();
-            $types[$parameterOrProperty->getName()] =
-                array_unique(
-                    $types[$parameterOrProperty->getName()]
-                );
+            if($unionType instanceof ReflectionNamedType) {
+                $types[$parameterOrProperty->getName()][]
+                    = $unionType->getName();
+                $types[$parameterOrProperty->getName()] =
+                    array_unique(
+                        $types[$parameterOrProperty->getName()]
+                    );
+            }
         }
         if(
             !in_array(
@@ -464,7 +465,7 @@ trait ReflectionTestTrait
      *                                           by the specified
      *                                           $reflectionNamedType.
      *
-     * @param array<string, array<int, string>> &$types
+     * @param array<string, array<int, string>> $types
      *                                             The array of $types
      *                                             to add the array of
      *                                             strings indicating

